### DISCUSSION TITLE
Welcome to gfg-potd Discussions!

### DISCUSSION THREAD

**Discussion by getlost01** [getlost01/gfg-potd#1] [Announcements]
### 👋 Welcome to the GitHub Discussion for discussing the "GFG Problem of the Day" solutions! 🚀

- We are excited to have you join this community of problem solvers and enthusiasts. This platform serves as a hub for insightful discussions, innovative approaches, and collaborative learning.

- Here, we encourage everyone to share their thoughts, ideas, and solutions related to the daily problems provided by GFG (GeeksforGeeks). Whether you are a beginner or an experienced coder, this is the perfect space to engage with fellow programmers, exchange knowledge, and explore different perspectives.

- Feel free to start a new thread or jump into existing discussions. Don't hesitate to ask questions, seek clarification, or offer your own solutions. Remember, this is a supportive environment where we value curiosity, respect diverse opinions, and foster a spirit of collaboration.

Happy coding, and enjoy the discussions ahead! 🎉

**Comment by chiranjib2001** [comment]
for today's(18-9-23) solution bruteforce alternative:
        long long p=1;
        for(long i=0;p<=n;p*=2){
            if(p==n) return true;
        }
        return false;

**Reply by getlost01** [reply]
Yeah thanks for your approach, but today I have created a editoral which is more optimize one  with O(1) complexity.

BTW my motive is to post as possible as optimise and non conventional solutions for better learnings.

Yeah once again thanks for posting your solution on discussion.

**Comment by chiranjib2001** [comment]
I solved today's(19-09-23) problem by this method: Your solution is optimal but I think I should share an alternate solution too. 
unsigned int getFirstSetBit(int n)
    {
        // Your code here
        int t=1;
        if(n==0) return 0;
        while(n%2==0){
            t++;
            n/=2;
        }
        return t;
    }

**Reply by getlost01** [reply]
Hello, I appreciate your contributions to our discussion section by sharing solutions. 
However, moving forward, could you please initiate a new discussion with a `Solution Idea` tag?
In this way, more people can benefit from your insights. 
Posting solutions in the announcement section may not receive as much attention, as individuals typically check for new discussions only if they are actively participating in the discussion threads.

**Comment by flyron21** [comment]
in today's question, we get segmentation error for the test case,
n=4
10 10 20 20
So i think using set or map is necessary to get correct answer.
What do you say?

**Reply by getlost01** [reply]
Hi! As the question states, we have only `0 to n-1` values in the array, so the `n` size array is enough to count the frequency.

<img width="513" alt="Screenshot 2023-09-24 at 2 19 22 PM" src="https://github.com/getlost01/gfg-potd/assets/79409258/d13f040c-fd67-41c4-ac0d-f5737068148b">

**Reply by flyron21** [reply]
okay, it is fine then, thanks for the reply, appreciate it.

**Comment by AmanGupta4251** [comment]
class Solution{
  public:
  vector<vector<int>> sumZeroMatrix(vector<vector<int>> a){
      //Code Here
      int n=a.size(),m=a[0].size();
      int area=0,x1=-1,x2=-1,y1=-1,y2=-1;
      for(int l=0;l<m;l++){
          vector<int>arr(n);
          for(int r=l;r<m;r++){
              for(int i=0;i<n;i++){
                  arr[i]+=a[i][r];
              }
              unordered_map<int,int>mp;mp[0]=-1;
              int sum=0,len=0,s=-1,e=-1;
              for(int i=0;i<n;i++){
                  sum+=arr[i];
                  if(mp.find(sum)!=mp.end()){
                     if(len<i-mp[sum]){
                         len=i-mp[sum];
                         e=i;
                         s=mp[sum]+1;
                     }
                  }
                  else mp[sum]=i;
              }
              if(area<len*(r-l+1)){
                  area=len*(r-l+1);
                  y1=l;y2=r;
                  x1=s;x2=e;
              }
          }
      }
      vector<vector<int>> ans;
      if(x1==-1||x2==-1||y1==-1||y2==-1) return ans;
      for(int i=x1;i<=x2;i++){
          vector<int>temp;
          for(int j=y1;j<=y2;j++){
              temp.push_back(a[i][j]);
          }
          ans.push_back(temp);
      }
      return ans;
  }
};



Here you go... I have solved GFG POTD for 26 December, 2023

**Reply by AmanGupta4251** [reply]
@getlost01

**Reply by getlost01** [reply]
Thank you for sharing your solution. It is a really good one.
